aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2270,6 +2281,47 @@ impl AptosVM {
             },
         })
     }
+
+    pub fn get_required_deposit(
+        &self,
+        session: &mut SessionExt,
+        resolver: &impl AptosMoveResolver,
+        txn_gas_params: &TransactionGasParameters,
+        txn_metadata: &TransactionMetadata,
+        payload: &TransactionPayload,
+    ) -> Option<u64> {
+        match payload {
+            TransactionPayload::EntryFunction(entry_func) => {
+                let has_randomness_attr =
+                    has_randomness_attribute(resolver, session, entry_func).unwrap_or(false);
+                if self.randomness_enabled && has_randomness_attr {
+                    let internal_gas_per_gas = u64::from(txn_gas_params.scaling_factor());
+                    let max_execution_gas = Gas::from(
+                        u64::from(txn_gas_params.max_execution_gas).div_ceil(internal_gas_per_gas),
+                    );
```

Comment:
@vgao1996 is there a better way to convert Gas to InternalGas?

Reply:
maybe just 
```
                    let max_execution_gas: Gas = txn_gas_params
                        .max_execution_gas
                        .to_unit_round_up_with_params(txn_gas_params);
```

Reply:
fixed